components/AdditiveCard/index.js

```diff
@@ -0,0 +1,79 @@
+import { atom, useAtom } from "jotai";
```

Comment:
Your AdditiveCard and you AllergenCard can be implemented as just one component, since they are behaving in the same way - they are just working on two different arrays. You can implement it like this:
```
import { useState } from "react";

export default function Card({ items }) {
  const [searchInput, setSearchInput] = useState("");
  const [selectedItems, setSelectedItems] = useState([]);

  const filteredItems = items.tags.filter(
    (item) => item.name.toLowerCase().indexOf(searchInput.toLowerCase()) === 0
  );

  return (
    <>
      <form
        onSubmit={(event) => {
          event.preventDefault();
          const formData = new FormData(event.target);
          const data = Object.fromEntries(formData);
          setSearchInput("");
          const selectedItem = filteredItems.find(
            (item) => item.name === data.search
          );
          if (
            //makes reselection impossible
            selectedItem &&
            !selectedItems.find((item) => item.name === selectedItem.name)
          )
            setSelectedItems([...selectedItems, selectedItem]);
        }}
      >
        <input
          name="search"
          type="text"
          value={searchInput}
          onChange={(event) => {
            setSearchInput(event.target.value);
          }}
        />
        <button type="submit">Add</button>
      </form>
      {filteredItems.length === 0 && searchInput.length > 0
        ? "No search results"
        : null}
      {searchInput.length > 0 && (
        <ul style={{ position: "relative" }}>
          {filteredItems.map((item) => {
            return (
              <div
                key={item.name}
                style={{
                  position: "absolut",
                  marginLeft: "-40px",
                  color: "grey",
                  maxWidth: "150px",
                }}
                onClick={() => {
                  setSearchInput(item.name);
                }}
              >
                {item.name}
              </div>
            );
          })}
        </ul>
      )}
      <ul style={{ listStyle: "none" }}>
        {selectedItems.map((selectedItem) => {
          return <li key={selectedItem.name}>{selectedItem.name} </li>;
        })}
      </ul>
    </>
  );
}
```

You can then use the component like this:
```
      {showAdditives ? (
        <Card key="additives" items={additives} />
      ) : (
        <Card key="allergens" items={allergens} />
      )}
```
Please note that in this special case you need to add a the key prop as shown above.

Reply:
this is not completly the same  in the additive component there is in line 16 in includes() function and in the allergen card component it is the indexOf() function